src/Colopl/Spanner/Concerns/ManagesTransactions.php

```diff
@@ -158,7 +161,14 @@ protected function performRollBack($toLevel)
 
         if ($this->currentTransaction !== null) {
             if ($this->currentTransaction->state() === Transaction::STATE_ACTIVE) {
-                $this->currentTransaction->rollBack();
+                try {
+                    $this->currentTransaction->rollBack();
+                } catch (NotFoundException $e) {
+                    // ignore session not found error
```

Comment:
Wouldn't ignoring the error cause unexpected behaviour upstream?

Reply:
this is made specially to allow to run rollback without error
```
				} catch (Throwable $e) {
					$this->rollBack();
```
otherwise there is dangling transaction  and everything is breaking 
there is test for such use-case - testInTransactionSessionNotFoundHandledError

in more general sense - if session if lost, there are no ways to rollback transaction at all, the only manageable way - quietly abort current transaction and then rerun everything again

Reply:
Thank you. That makes sense. Could you add that last sentence as a comment?

Reply:
sure, also I found edge case when exception anyway should be called - i.e. on explicit rollback inside transaction({}), 
I am adding test-case and code for handling

Reply:
done, please check